packages/swap/src/ingress-egress-tracking/index.ts

```diff
@@ -56,9 +56,18 @@ export const getPendingDeposit = async (
       return mempoolTx && { ...mempoolTx, transactionConfirmations: 0 };
     }
 
+    const currentHeight = (await prisma.state.findFirstOrThrow()).height;
+    // For boosted swaps, state chain will track the foreign chain block one state chain block before the DepositBoosted event
+    // Because of this, frontend will jump to confirmations page and then, will switch back to checking boost liquidity
+    // To prevent this, we need to wait for 1 state chain blocks before checking the confirmations
+    const isTrackingPassedTwoStateChainBlocks =
+      currentHeight - (tracking?.blockTrackedAtStateChainBlock ?? 0) > 0;
+
```

Comment:
then we could tell the story why this is needed somehow like this:
```
// deposits in an external block will be witnessed (and boosted) one statechain block after chain tracking is updated
// to prevent jumping from "1 confirmation" to "checking boost liquidity", we need to calculate confirmations based on the last witnessed block
const statechainBlocksSinceTracking = currentHeight - (tracking?.blockTrackedAtStateChainBlock ?? 0);
const lastWitnessedBlockHeight = statechainBlocksSinceTracking > 0 ? tracking.height : tracking.previousHeight

...
confirmations = statechainBlocksSinceTracking - deposits[0].deposit_chain_block_height + 1
```